src/kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -75,24 +75,18 @@ impl<'tcx> GotocCtx<'tcx> {
         loc: Location,
     ) -> Stmt {
         assert!(cond.typ().is_bool());
-
         let property_name = property_class.as_str();
-
-        // Create a Property Location Variant from any given Location type
-        let property_location =
-            Location::create_location_with_property(message, property_name, loc);
-
-        Stmt::assert(cond, property_name, message, property_location)
+        Stmt::assert(cond, property_name, message, loc)
```

Comment:
Ah, maybe this answers my previous question about root cause. So we went from applying it early, to applying it in a later choke point (with asserts looking for reasonable values), to ensure we don't forget to do so. Right? Seems like an excellent change. Nice.

Reply:
Yeah, that's about it. It wasn't consistently applied everywhere which created the issue for a few `assert_false` calls. Now, `assert` all ensure structure and quality of the values that it needs.